enhancements/csvless-bundles.md

```diff
@@ -24,145 +24,452 @@ status: implementable
+    path: spec.deployment.view_only_mode
+    capabilities:
+    - 'urn:alm:descriptor:com.tectonic.ui:booleanSwitch'
+  - displayName: Web Root
+    description: "Defines the root context path for the Kiali console, API endpoints and readiness/liveness probes. Default: / (when deployed on OpenShift; /kiali (when deployed on Kubernetes)"
+    path: spec.server.web_root
+    capabilities:
+    - 'urn:alm:descriptor:com.tectonic.ui:label'
+- apiVersion: monitoring.kiali.io/v1alpha1
+  kind: MonitoringDashboard
+  description: A configuration file for defining an individual metric dashboard.
+  displayName: Monitoring Dashboard
+  descriptors:
+  - displayName: Title
+    description: "The title of the dashboard."
+    path: title
+    capabilities:
+    - 'urn:alm:descriptor:com.tectonic.ui:label'
 ```
 
-The `apis` element is a special case and is present to enable to organization of [descriptors](https://github.com/operator-framework/operator-lifecycle-manager#descriptors). It will be kept up to date with `provided APIs` inferred by the `opm bundle build` command upon execution.
+_**Note:** `capabilities` is the equivalent of the `x-descriptors` field found in CSVs._
+_**Note:** The `path` field of a `descriptor` informs its "descriptor type" at install time. Only `.spec` and `.status` root paths are allowed and would map respectively to spec and status descriptors of an equivalent CSV. What was known as action descriptors -- with respect to CSVs -- are not supported, and in most cases can be exchanged for an equivalent spec or status descriptor anyway._
 
-_**Note:** The `descriptors.path` element's value is qualified with the `status` subresource. For comparision, a CSV contains two types of descriptors -- one for spec, the other for status -- both with path elements that don't contain this additional qualifier._
+The `annotations.yaml` file contains the set of labels to be projected onto the container image representation of the bundle. More importantly, it contains the top-level discriminator field used to identify the mediatype of the bundle on-disk: `operators.operatorframework.io.bundle.mediatype: registry+v2`.
 
-_**Note:** `capabilities` is the equivalent of the `x-descriptors` field found in CSVs._
+#### Build a Bundle Image Without a CSV
 
-### User Stories
+As an Operator Author, I want to build an Operator Bundle image from the same set of Kubernetes manifests I would use to deploy an operator manually, without OLM, so that I can avoid writing a CSV.
 
-#### Packaging an Operator Without a CSV
+##### UX
 
-As an Operator Author, Alice wants to package her operator for use with OLM without writing a CSV, so that she can avoid duplicating deployment configuration that can be expressed with standard Kubernetes resources.
+Add a new command to `opm`, `opm bundle init`,  to initialize the current directory as a bundle, which has:
```

Comment:
`opm alpha bundle generate` was the original target. The existing UX for it -- in part, producing a valid output bundle -- conflicted with the UX for `registry+v2` (because we can't expect metadata `registry+v2` input manifests.

The other side of things was that I wanted to take a pass at making the command more uniform wrt to other tools I've seen.

If we had a concrete support statement -- i.e. no support -- for `alpha` subcommands, I would be inclined to change the `generate` command in a non-backwards compatible way. 

>  I think we should discuss if having two obviously visible ways to do this makes sense

I too would like to avoid fracturing the UX between projects. I think we should expose everything in both -- at different rates -- or stick with a single project exclusively. We could always contribute these commands directly to SDK, but I'm concerned that we'd end up reducing the rate at which we iterate bundle UX as a result. 

My understanding was that opm was more or less meant to be our tightly-scoped testbed/plumbing, and SDK was meant to be the porcelain that normalizes the experience. 

Alternatively, we could say that on-disk scaffolding is outside the scope of the "plumbing commands", and drop this -- and generate -- entirely.

Reply:
> Alternatively, we could say that on-disk scaffolding is outside the scope of the "plumbing commands", and drop this -- and generate -- entirely.

I am a +1 on this idea.

>  but I'm concerned that we'd end up reducing the rate at which we iterate bundle UX as a result.

soon we need to get to the point where this is the case IMHO. We need to start stabilizing the "new/correct" way to interact with this stuff. While we have done a good job of being backward compatibles I think that we need to start slowing down the pace of change.